docs/jest-enzyme-matchers.md

```diff
@@ -0,0 +1,76 @@
+# Require the use of enzyme matchers when possible (jest-enzyme-matchers)
+
+[jest-enzyme](https://github.com/FormidableLabs/enzyme-matchers/tree/master/packages/jest-enzyme#readme)
+provides a number of matchers that useful when writing enzyme tests.  Using these
+matchers providers better error messages when there are test failures.  This rule
+supporters auto-fixing.
+
+## Rule Details
+
+The following are considered warnings:
+
+```js
+expect(wrapper.first().prop("foo")).toEqual("bar");
+```
+
+```js
+expect(wrapper.first().state("foo")).toEqual("bar");
+```
+
+```js
+expect(wrapper.find(".foo")).toHaveLength(2);
+```
+
+```js
+expect(wrapper.find(".foo").text()).toEqual("bar");
+```
+
+```js
+expect(wrapper.find(".foo").html()).toEqual("<p>bar</p>");
+```
+
+```js
+expect(wrapper.find(".foo").exists()).toBeTrue();
+```
+
+```js
+expect(wrapper.exists(".foo")).toBeTrue();
+```
+
+```js
+expect(wrapper.find(".foo").exists()).toBeFalse();
+```
+
+```js
+expect(wrapper.exists(".foo")).toBeFalse();
+```
+
+The following are not considered warnings:
```

Comment:
I found it to be a little challenging to figure out which was the "correct" version of each of the above. It might be convenient to format it like:

```
// Invalid
expect(wrapper.first().prop("foo")).toEqual("bar");
// Valid
expect(wrapper.first()).toHaveProp("foo", "bar");
```

Reply:
That's a good point.  I've been following how official rules present the material, but showing invalid/valid pairings, especially when there's a number of different cases makes a lot of sense.